pkg/kops/kops.go

```diff
@@ -89,3 +89,23 @@ func GetAutoScalingGroupNameFromKopsMachinePool(kmp kinfrastructurev1alpha1.Kops
 
 	return &asgName, nil
 }
+
+func GetVirtualNodeGroupNameFromKopsMachinePool(kmp kinfrastructurev1alpha1.KopsMachinePool) (*string, error) {
```

Comment:
Couldn't we refactor the GetAutoScalingGroupNameFromKopsMachinePool to be more generic and do the same thing as this one as well?

Reply:
Done, merged both as GetCloudResourceNameFromKopsMachinePool